metrics-publishers/cloudwatch-metrics-publisher/src/main/java/software/amazon/awssdk/metrics/publishers/cloudwatch/internal/MetricConsumer.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics.publishers.cloudwatch.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.cloudwatch.model.MetricDatum;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataRequest;
+import software.amazon.awssdk.services.cloudwatch.model.PutMetricDataResponse;
+import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Retrieves {@link MetricDatum}s from the shared blocking queue and upload them to Amazon CloudWatch.
+ */
+@SdkInternalApi
+public class MetricConsumer implements Callable<CompletableFuture<PutMetricDataResponse>> {
+
+    private static final Logger log = Logger.loggerFor(MetricConsumer.class);
+
+    /**
+     * CloudWatch limits number of {@link MetricDatum} to 20 per {@link PutMetricDataRequest}.
+     */
+    private static final int MAX_METRIC_PER_REQUEST = 20;
+
+    private final CloudWatchAsyncClient client;
+    private final BlockingQueue<MetricDatum> queue;
+    private final String namespace;
+
+    private MetricConsumer(Builder builder) {
+        this.client = Validate.notNull(builder.cloudWatchClient, "CloudWatch client cannot be null");
```

Comment:
nit, we can use `Validate.paramNotNull(builder.cloudWatchClient, "cloudWatchClient")`

Reply:
Updated